### DISCUSSION TITLE
TDS: IEEE14 bus system, Use Case: 1.2 demo_DGPRCTExt.ipynb

### DISCUSSION THREAD

**Discussion by JunejaAmmy** [CURENT/andes#251] [General]
Good morning Dr. Cui,

I followed the **1.2 demo_DGPRCTExt.ipynb** use case and it generated below output files:
1. ieee14_dgprctext_out.lst
2. ieee14_dgprctext_out.npz

As per further instructions, I updated the external voltage once run the below command:
`ss.DGPRCTExt.set(src='v', idx='DGPRCTExt_1', attr='v', value=0.7)`

It ran successfully but it generated exactly the same output files.

My queries are:
1. Are these the only output files which generated during simulation (TDS)?
2. There was no impact on output files, after changing the external voltage (as mentioned in use case), Is that correct?
3. Please guide me if I am missing something?

**Comment by cuihantao** [comment]
@jinningwang Could you follow up on the conversation?

**Reply by jinningwang** [reply]
Sure, let me take a look.

Regards,
Jinning

**Comment by jinningwang** [comment]
@SinghAS18 
Hi there,

You said you get the same results, and I think it is reasonable. It is because the original output file is generated under the condition that ``DGPRCTExt_1.v = 0.7``. Therefore, you will definitely get the same output file when you run the command.

You can check the parameters of DGPRCTExt (the input excel file) and modify the voltage thresholds. Then you can set different voltage values in that command to trigger the protection.

Hope this can help.

Regards,
Jinning

**Reply by JunejaAmmy** [reply]
Thank you for your response,

I went through the sheet "**DGPRCTExt**" under use case "**ieee14_dgprctext.xlsx**" and found these header names "**vl4, vl3, vl2, vl1, vu1, vu2, vu3**" which referred to voltage.

We further gone through online documentation of ANDES and found vl4 to vl1 are referred to "Under voltage" and vu1 to vu3 are referred to "Over voltage".

Please suggest whether above ones are related to "voltage thresholds" as per your response else suggest the correct "voltage thresholds" values.

We updated the above mentioned under voltage and over voltage values in input excel file and run the Power flow and then TDS commands, but there was no change in below output files:
1. ieee14_dgprctext_out.lst
2. ieee14_dgprctext_out.npz
3. ieee14_dgprctext_out.txt

**Queries**:
1. Are these the only output files to check/view the output results?
2. Please suggest the correct valid range of under voltage and over voltage values.

**Reply by jinningwang** [reply]
Q1: You can plot from those files. Also, you can directly plot the results in the notebook file. Please refer to the example files and check out how to visualize the simulation results.
Q2: In the [documentation](https://docs.andes.app/en/stable/groupdoc/DGProtection.html?highlight=dgprctext#dgprctext), you may find the settings as follow:
Frequency (Hz):

(fl3, fl2), Tfl2 [(50.0, 57.5), 10s]

(fl2, fl1), Tfl1 [(57.5, 59.2), 300s]

(fu1, fu2), Tfu1 [(60.5, 61.5), 300s]

(fu2, fu3), Tfu2 [(61.5, 70.0), 10s]

Voltage (p.u.):

(vl4, vl3), Tvl3 [(0.10, 0.45), 0.16s]

(vl3, vl2), Tvl2 [(0.45, 0.60), 1s]

(vl2, vl1), Tvl1 [(0.60, 0.88), 2s]

(vu1, vu2), Tvu1 [(1.10, 1.20), 1s]

(vu2, vu3), Tvu2 [(1.20, 2.00), 0.16s]

The protection is designed to be triggered ONLY when the target signal lasts for the given time. For example, if the voltage falls into (vl2, vl1) and lasts for 2s, the protection will be triggered. For more details, you can check NERC. Bulk Power System Reliability Perspectives on the Adoption of IEEE 1547-2018. March 2020.

Regards,
Jinning
